app/lib/ordered_result_calculator.rb

```diff
@@ -33,68 +33,82 @@ def update_age_group_results
-        Result.create_new!(competitor, new_place, "AgeGroup", competitor.age_group_entry_description) if competitor.ineligible?
-      end
-
-      # score the eligibles
-      competitors_in_sorted_order = CompetitorOrderer.new(selected_competitors, lower_is_better, use_with_ineligible_score: false).sort
-      competitors_in_sorted_order = competitors_in_sorted_order.reject(&:ineligible?)
-      competitors_in_sorted_order.each do |competitor|
-        gender_place_calc = get_place_calculator("AgeGroup Eligible: #{competitor.age_group_entry_description}")
-        new_place = gender_place_calc.place_next(competitor.comparable_score, tie_break_points: competitor.comparable_tie_break_score, dq: competitor.disqualified?)
-        Result.create_new!(competitor, new_place, "AgeGroup", competitor.age_group_entry_description)
-      end
+      placed_ineligible_competitors = compute_places_of_ineligible_competitors(competitors_in_sorted_order_with_ineligible, ->(competitor) { "AgeGroup Ineligible: #{competitor.age_group_entry_description}" })
+      placed_eligible_competitors = compute_places_of_eligible_competitors(selected_competitors, ->(competitor) { "AgeGroup Eligible: #{competitor.age_group_entry_description}" })
+
+      placed_ineligible_competitors.concat(placed_eligible_competitors)
     end
   end
 
-  def all_competitors_in_sorted_order
-    @all_competitors_in_sorted_order ||= competitors_in_sorted_order(competitors)
+  def compute_places_of_competitors_including_ineligible(competitors_in_sorted_order_with_ineligible, group_name_provider)
+    competitors_in_sorted_order_with_ineligible.map do |competitor|
+      compute_place_of_competitor(competitor, group_name_provider.call(competitor), true)
+    end
   end
 
-  def competitors_in_sorted_order(selected_competitors)
-    @competitors_in_sorted_order ||= CompetitorOrderer.new(selected_competitors, lower_is_better).sort
+  def compute_places_of_ineligible_competitors(competitors_in_sorted_order_with_ineligible, group_name_provider)
+    competitors_in_sorted_order_with_ineligible.map do |competitor|
+      compute_place_of_competitor(competitor, group_name_provider.call(competitor), true, only_ineligible: true)
+    end
+  end
+
+  def compute_places_of_eligible_competitors(competitors, group_name_provider)
+    competitors_in_sorted_order = CompetitorOrderer.new(competitors, lower_is_better, use_with_ineligible_score: false).sort
+    competitors_in_sorted_order = competitors_in_sorted_order.reject(&:ineligible?)
+    competitors_in_sorted_order.map do |competitor|
+      compute_place_of_competitor(competitor, group_name_provider.call(competitor), false)
+    end
+  end
+
+  def compute_place_of_competitor(competitor, age_group_desc, score_with_ineligible, only_ineligible: false)
```

Comment:
Please don't use positional arguments for true/false statements (`score_with_ineligible`....though this isn't a great name...I know it's my fault)). In general, when using multiple arguments (not just 1), using named-arguments makes it much easier to understand what is happening when looking at the `calling` code, instead of having to referenced the `called` code to understand what `true` or `false` means.